File: Boucles/TP4_11.js
function occurence(array) {
  let i = 0;
  let j;
  let control1;
  let control2;
  let array2 = [];
  let presence = false;
  while (i < (array.length)) {
    let control1 = array[i];
    j = 0;
    console.log("search " + control1);
    while (j < (array2.length)) {
      let control2 = array2[j];
      if (control1 == control2) {
        presence = true;
        console.log("find " + control1);
      }
      j++;
    }
    if (!presence) {
      console.log("push " + control1);
      array2.push(control1);
    }
    presence = false;
    i++;
  }
  return array2;
}

let test = [0, 5, 5, 4, 4, 3, 3, 1, 1, 2, 2, 2, 3, 4, 5, 5, 6];
let result = occurence(test);
console.log(result);

File: Boucles/TP4_11.py
def occurence(array) :
    i = 0;
    array2 = [];
    presence = False;
    while i < len(array) :
        control1 = array[i];
        j = 0;
        while j < len(array2) :
            control2 = array2[j];
            if (control1 == control2) :
                presence = True;
            j+=1;
        if (presence == False) :
            array2.append(control1);
        presence = False;
        i+=1;
    return array2

test = [0, 5, 5, 4, 4, 3, 3, 1, 1, 2, 2, 2, 3, 4, 5, 5, 6];
result = occurence(test);
print(result);
